packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -826,15 +826,11 @@ class RenderParagraph extends RenderBox with ContainerRenderObjectMixin<RenderBo
             locale: locale,
           )..layout();
           if (didOverflowWidth) {
-            double fadeEnd, fadeStart;
-            switch (textDirection) {
-              case TextDirection.rtl:
-                fadeEnd = 0.0;
-                fadeStart = fadeSizePainter.width;
-              case TextDirection.ltr:
-                fadeEnd = size.width;
-                fadeStart = fadeEnd - fadeSizePainter.width;
-            }
+            double fadeStart, fadeEnd;
+            (fadeStart, fadeEnd) = switch (textDirection) {
```

Comment:
Could we assign directly here?

Reply:
Unfortunately I don't think it's possible.

```dart
final (double fadeStart, double fadeEnd) = // works great for finals
(double fadeStart, double fadeEnd) = // syntax error
var (double fadeStart, double fadeEnd) = // triggers the "don't use var" lint
```

Reply:
Are `fadeStart` and `fadeEnd` ever reassigned again?

Reply:
ooh, you're right, thanks!